trendy-tracker.kr/src/components/Twemoji.tsx

```diff
@@ -0,0 +1,34 @@
+import Image from 'next/image';
+import { memo } from 'react';
+import twemoji from 'twemoji';
+
+const U200D = String.fromCharCode(0x200d);
+const UFE0Fg = /\uFE0F/g;
+
+function Twemoji({
+  emoji,
+  ext = 'svg',
+  width = 72,
+  height = 72,
+}: {
+  emoji: string
+  ext?: 'svg' | 'png'
+  width?: number | string
+  height?: number | string
+}) {
+  const HEXCodePoint = twemoji.convert.toCodePoint(
+    emoji.indexOf(U200D) < 0 ? emoji.replace(UFE0Fg, '') : emoji,
+  );
```

Comment:
https://github.com/twitter/twemoji/issues/419#issuecomment-637360325

오래된 이모지를 지원하기위한 normalize 코드조각

```
In the source code you can find how the image name is retrieved, which boils down to:

var UFE0Fg = /\uFE0F/g;
var U200D = String.fromCharCode(0x200D);
var toCodePoint = twemoji.convert.toCodePoint;
/**
 * Used to both remove the possible variant
 *  and to convert utf16 into code points.
 *  If there is a zero-width-joiner (U+200D), leave the variants in.
 * @param   string    the raw text of the emoji match
 * @return  string    the code point
 */
function grabTheRightIcon(rawText) {
  // if variant is present as \uFE0F
  return toCodePoint(rawText.indexOf(U200D) < 0 ?
    rawText.replace(UFE0Fg, '') :
    rawText
  );
}
There is eventually a twemoji.convert.fromCodePoint to rollback.

The \uFE0F is removed because old emoji are inconsistently implemented across browsers, were few might, and few others might not, consider/use that char, so the function tries to normalize/work for all browsers.
```